contracts/clients/CkbProof.sol

```diff
@@ -0,0 +1,365 @@
+    RLPReader.RLPItem[] memory commitmentsRlp = ls[1].toList();
+    CommitmentKV[] memory commitments;
+    for (uint i = 0; i < commitmentsRlp.length; i++) {
+        RLPReader.RLPItem[] memory kvRlp = commitmentsRlp[i].toList();
+        commitments[i] = CommitmentKV(kvRlp[0].toUint(), kvRlp[1].toUint());
+    }
+
+    // Decode the content
+    bytes memory content = ls[2].toBytes();
+
+    // Return the decoded Envelope
+    return Envelope(msg_type, commitments, content);
+}
+
+function parseCommitment(
+    bytes memory ckbTransaction
+) pure returns (CommitmentKV[] memory) {
+    uint256 witness_count = ckbTransaction.readCKBTxWitnessCount();
+    uint8 output_type_index = 2;
+    (uint256 offset, uint256 size) = ckbTransaction.readCKBTxWitness(
+        uint8(witness_count),
+        output_type_index
+    );
+    bytes memory output_type_bytes = new bytes(size);
+    for (uint i = 0; i < size; i++) {
+        output_type_bytes[i] = ckbTransaction[i + offset];
+    }
+    Envelope memory witness_struct = decodeRlpEnvelope(output_type_bytes);
+    return witness_struct.commitments;
+}
+
+function verifyHashExist(
+    bytes32[] memory leaves,
+    bytes32 witnessHash
+) pure returns (bool) {
+    bool isInLeaves = false;
+    for (uint i = 0; i < leaves.length; i++) {
+        if (leaves[i] == witnessHash) {
+            isInLeaves = true;
+            break;
+        }
+    }
+
+    return isInLeaves;
+}
+
+function isCommitInCommitments(
+    CommitmentKV[] memory commitments,
+    bytes memory key,
+    bytes calldata value
+) pure returns (bool) {
+    uint256 keyHash = uint256(keccak256(key));
+    uint256 valueHash = uint256(keccak256(value));
+    for (uint i = 0; i < commitments.length; i++) {
+        if (
+            commitments[i].key == keyHash && commitments[i].value == valueHash
+        ) {
+            return true;
+        }
+    }
+    return false;
+}
+
+import "truffle/console.sol";
+
+library CkbProof {
+    event Log(string message, uint value);
+    function decodeTestProof(
+        bytes memory rlpData
+    ) public pure returns (TestProof memory) {
+        RLPReader.RLPItem[] memory items = RLPReader.toList(
+            RLPReader.toRlpItem(rlpData)
+        );
+
+        TestProof memory axonProof;
+        axonProof.ckbTransaction = RLPReader.toBytes(items[0]);
+
+        return axonProof;
+    }
+
+    function decodeAxonObjectProof(
+        bytes memory rlpData
+    ) public pure returns (AxonObjectProof memory) {
+        RLPReader.RLPItem[] memory items = RLPReader.toList(
+            RLPReader.toRlpItem(rlpData)
+        );
+
+        AxonObjectProof memory axonProof;
+        axonProof.ckbTransaction = RLPReader.toBytes(items[0]);
+        axonProof.blockHash = bytes32(RLPReader.toBytes(items[1]));
+
+        bytes memory payloadBytes = RLPReader.toBytes(items[2]);
+        RLPReader.RLPItem[] memory payloadItems = RLPReader.toList(
+            RLPReader.toRlpItem(payloadBytes)
+        );
```

Comment:
Should be `RLPReader.RLPItem[] memory payloadItems = RLPReader.toList(items[2])`